kyo-combinators/shared/src/test/scala/kyo/EnvCombinatorTest.scala

```diff
@@ -57,6 +57,24 @@ class EnvCombinatorTest extends Test:
                         .provideLayer(layerBool)
                 assert(Memo.run(handled).eval == 23)
             }
+
+            "should provide all layers" in {
```

Comment:
Can you add a tests checking that `provide` and `Env.runLayer` infers the expected pending effects? I normally do:

```scala
val x = ...
val y: ExpectedType = x
```

Reply:
I added them. Do you want them in separate tests?

Reply:
that looks good, thanks!